applications/registration-api/src/main/java/no/dcat/service/permission/ResourceRoleFactory.java

```diff
@@ -0,0 +1,16 @@
+package no.dcat.service.permission;
+
+class ResourceRoleFactory {
+    static ResourceRole deserialize(String roleToken) {
+        String[] parts = roleToken.split(":");
+        String resourceType = parts[0];
+        String resourceId = parts[1];
+        String resourceRole = parts[2];
```

Comment:
As long as we can guarantee that parts array always has three elements, this is ok. What if it doesn't?

Reply:
Da skal det kaste en runtime error. Hvilke error liker du best?